posts/2023-10-20-open-liberty-with-cargo-tracker-ee10.adoc

```diff
@@ -0,0 +1,119 @@
+
+Cargo Tracker books, tracks, updates, and routes cargo objects. Messaging events are logged to update cargo locations, while persistence ensures that cargo status is not changed or removed when the application is shut down. Each portion of Cargo Tracker represents a step in a real-life shipment process: the customer tracks their cargo using the Public Tracking interface, the shipping company schedules and routes deliveries using the Administration interface, and port handlers log arriving and departing cargo using the Mobile Event Logger. 
+
+image::/img/blog/cargo-tracker-ee10-home.png[Cargo Tracker Home Page,width=50%,align="left"]
+image::/img/blog/cargo-tracker-ee10-map.png[Cargo Tracker Map,width=50%,align="right"]
+
+
+== Run Cargo Tracker with Open Liberty 
+
+Cargo Tracker EE10 supports Java SE 11 and 17. https://developer.ibm.com/languages/java/semeru-runtimes/downloads[IBM Semeru Runtime] is the preferred Java runtime when using Open Liberty. Before you clone the application to your machine, install any required JDKs and ensure that your `JAVA_HOME` system variable is set to a compatible version. To access the repository remotely you'll need to install https://git-scm.com/book/en/v2/Getting-Started-Installing-Git[Git] if you haven't already. Finally, you will need to install https://maven.apache.org/download.cgi[Maven] to build and run the application. 
+
+You can clone the Cargo Tracker project to your local machine by running:
+
+[source]
+----
+git clone https://github.com/eclipse-ee4j/cargotracker.git
+----
+
+You can also visit the https://github.com/eclipse-ee4j/cargotracker/tree/master[Git repository the Cargo Tracker project] to take a peek at the code.
+
+To get started, choose one of the following setup methods:
+
+* <<maven_setup, Maven command line setup>> 
+* <<vscode_setup, Visual Studio Code IDE setup>>
+
+https://openliberty.io/docs/latest/development-mode.html[Liberty dev mode] is used in either setup because it provides three key features. Automatically detect, recompile, and deploy code changes in your running Liberty. Unit and integration tests are run on demand when you press `Enter` in the command terminal. Finally, you can attach a debugger to the running Liberty to step through the code. Through those features, you can deeply learn the Jarakee EE through debugging or updating the Cargo Tracker code.
+
+
+[#maven_setup]
+== Maven command line setup
+
+After cloning the project, you can quickly start Cargo Tracker through the command line with Maven. Navigate to the project base directory (where the pom.xml file is located) and run:
+
+[source]
+----
+mvn -Popenliberty liberty:dev
+----
+ 
+The application should start without any thrown exceptions. You can safely ignore the shrinkwrap features warning, message-driven bean warning, and the AggregateObjectMapping nested foreign key warning, as they don't affect the application functionality.
```

Comment:
Can we link out to any external links that give further detail as to what these warnings are so if people are concerned they can look into them for themselves?

Reply:
I doubt we can

Reply:
ok, no worries then.